pandas/core/arrays/base.py

```diff
@@ -343,8 +338,7 @@ def astype(self, dtype, copy=True):
         """
         return np.array(self, dtype=dtype, copy=copy)
 
-    def isna(self):
-        # type: () -> Union[ExtensionArray, np.ndarray]
+    def isna(self) -> Union[ABCExtensionArray, np.ndarray]:
```

Comment:
so this is a the first stab at ArrayLike (e.g. to put into pandas.typing)

Reply:
Seems like a reasonable place to start.

Reply:
these are ok here for now, mainly adding things that we *should* add

Reply:
Agreed this would be great @gwrome do you mind opening up an issue as a follow up?

Reply:
No problem. I should be able to do that this weekend.